### PR TITLE
Fix missing Homebrew prefix resolution.

### PR DESCRIPTION
**Pull request by rkxx08**
Make package making script to work with arbitrary (non-default) Homebrew locations.

### CODE REVIEW

File: utils/make-pkgconfig.swift
Comment (matthewseaman) [inline utils/make-pkgconfig.swift]: We cannot assume the LLVM installation was done via `brew` because that breaks Linux users who didn't use `brew` to install (or Mac users who didn't use `brew` for that matter). This is why the CI is failing.

File: utils/make-pkgconfig.swift
Comment (matthewseaman) [inline utils/make-pkgconfig.swift]: Let's keep this in a closure since it only needs to be executed if the other 2 `which` calls on line 63 returned `nil`.

**Review by matthewseaman — CHANGES_REQUESTED** [review]

**Review by matthewseaman — CHANGES_REQUESTED** [review]
See above comments. Basically we can't assume that `brew` was used to install LLVM.

File: utils/make-pkgconfig.swift
Comment (rkxx08) [reply utils/make-pkgconfig.swift]: I see.
I'm going to add platform branching. (`#available`)

For the Linux, I don't know well about it. Just keep `/usr/local/` for Linux with hoping some Linux expert to come up to fix it.

For the macOS, I think `/usr/local` can cover only x86 Homebrew at default location. (maybe some other package managers?) How about calling `brew --prefix` first and falling back to `/usr/local`?

File: utils/make-pkgconfig.swift
Comment (rkxx08) [reply utils/make-pkgconfig.swift]: Okay

File: utils/make-pkgconfig.swift
Comment (matthewseaman) [reply utils/make-pkgconfig.swift]: I'm not sure how useful `#available` will be since [brew](https://brew.sh) is now available for Linux I think, so in theory someone could install it that way.

As for trying brew first then falling back to `/usr/local`, that's probably fine. @segiddins @dduan what do you think? Could still be a problem for those who have `brew` installed but installed it via some other means. Not sure how much we care about that.

**Comment by rkxx08** [comment]
@matthewseaman I pushed new commit. Thanks.

File: utils/make-pkgconfig.swift
Comment (rkxx08) [reply utils/make-pkgconfig.swift]: I agree to that `#available` is not necessary.

**Review by matthewseaman — APPROVED** [review]
Looks good to me, though @segiddins or @dduan should take a look before merging to make sure this is fine.

**Comment by rkxx08** [comment]
Okay

**Review by segiddins — APPROVED** [review]

**Comment by matthewseaman** [comment]
@eonil Thanks for this!

**Pull request merged**
